spec/integration/federation/factories.rb

```diff
@@ -0,0 +1,219 @@
+require "factory_girl"
+
+def r_str
+  SecureRandom.hex(3)
+end
+
+#
+# Sort hash according to an entity class's property sequence.
+# This is used for rspec tests in order to generate correct input hash to
+# compare results with.
+#
+def sort_hash(data, klass)
+  klass.class_props.map { |prop|
+    [prop[:name], data[prop[:name]]] unless data[prop[:name]].nil?
+  }.compact.to_h
+end
```

Comment:
The name of the method feels a bit too generic for this project. Also note that hashes compare order independent, `{a: 1, b: 2} ==  {b: 2, a: 1 } #=> true`